pytorch_lightning/core/lightning.py

```diff
@@ -1716,11 +1716,13 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
         elif self.example_input_array is not None:
             input_data = self.example_input_array
         else:
-            raise ValueError('`input_sample` and `example_input_array` tensors are both missing.')
-
+            raise ValueError('Could not export to ONNX since neither `input_sample` nor'
+                             ' `model.example_input_array` attribute is set.')
```

Comment:
I think this warning is not correct if input_sample is neither a Tensor nor None.

Reply:
so how would you change it?
I see the condition above and it states as the error say - if `input_sample` is not tensor then it is not valid and `model.example_input_array is None` means it is missing...

Reply:
```suggestion
            if input_sample is not None:
                raise ValueError('`input_sample` should be a Tensor')
            else:
                raise ValueError('Could not export to ONNX since neither `input_sample` nor'
                                 ' `model.example_input_array` attribute is set.')
```
I don't know, maybe this?

Reply:
then lest be more spec with the first one 
```
raise ValueError(f'passed `input_sample` if {type(input_sample)} but it should be a Tensor')
```

Reply:
@rohitgr7 good catch 👍 . I'll add the changes as @Borda suggested. :]